vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java

```diff
@@ -1295,33 +1295,29 @@ private static String cellText(int lineno, String s) {
 
     @Override
     public Component getListCellRendererComponent(
-      JList list, Object value, int index, boolean selected, boolean hasFocus) {
+            JList list, Object value, int index, boolean selected, boolean hasFocus) {
 
       // DO NOT pass value to super.getListCellRendererComponent()
       // It is incredibly inefficient for GamePieces and is not needed
       // since we overwrite the label text anyway.
       super.getListCellRendererComponent(list, "", index, selected, hasFocus);
 
-      final int lineOffset = definer.isPrototype() ? 1 : 0;
-      final int lineno = index + lineOffset;
+      // For Pieces, bump index to 1
+      final int lineno = definer.isPrototype() ? index + 1 : index;
 
       if (value instanceof EditablePiece) {
-        final EditablePiece editableValue = (EditablePiece) value;
-
-        // Prototype? Skip first (dummy) item to avoid outputting a blank line
-        if (lineOffset == 1 || index > 0) {
-          // For Pieces, bump index to 1
-          // Special formatting for Comments
-          if ((editableValue.getType().startsWith(Comment.ID))) {
-            setText(cellText(lineno, "<b>/* " + ConfigureTree.noHTML((editableValue.getDescription())) + " */</b>"));
-          }
-          else {
-            setText(cellText(lineno, ConfigureTree.noHTML(editableValue.getDescription())));
-          }
+
+        if (value instanceof Decorator && ((Decorator) value).myGetType().startsWith(Comment.ID)) {
+          setText(cellText(lineno, "<b>/* " + ConfigureTree.noHTML(((EditablePiece) value).getDescription() + " */")));
+        }
+        else if (lineno > 0) {
+          // Prototype? Skip first (dummy) item
+          setText(cellText(lineno, ConfigureTree.noHTML(((EditablePiece) value).getDescription())));
         }
+
       }
       else {
-        // FIXME: When does this get used ? Assumed that prototype might have to be skipped here also.
+        // Redundant ?
```

Comment:
I really don't think anything coming through can not be an EditablePiece, , I think the `instaceof EditablePiece` above is just defensive programming to prevent un-guarded casts to EditablePiece. Does no harm being there, I would leave it in case there are other case we're not aware of. Ah, just found one, comment below.

Reply:
PR #13001 modifies the comments to clarify the catch-all purpose of the EditablePiece test and related code.